### PR TITLE
fix(authentication-service):  added idp server controller for login and discovery endpoint

### PR DESCRIPTION
**Pull request by prernagp**
## Description

Added IDP server controller for login and discovery endpoint
 
BREAKING CHANGE
Replaced use of private and public keys from files to database. 
jwt_keys table has been introduced in the auth database schema.
asymmetric signer and verifier will use the keys from the database.
If keys are not present, asymmetric token signing and verification will not work.

Fixes # (issue) #2034 

## Type of change

Please delete options that are not relevant.

- [x] Breaking Change (breaking change which adds functionality)

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] Code conforms with the style guide
- [x] API Documentation in code was updated

### CODE REVIEW

**Review by sfvishalgupta — COMMENTED** [review]
I can not see all the endpoints mentioned here, https://docs.google.com/document/d/1r2v55eC5dbGBsA5ka_JQnzIIO80hdTqA6p8qCRKQmqU/edit
Also please specify and work on key rotation

File: services/authentication-service/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     }
   },
   "scripts": {
+    "start": "node -r source-map-support/register .",
```
Comment (samarpan-b) [inline services/authentication-service/package.json:28]: why this ?

File: services/authentication-service/src/modules/auth/controllers/identity-server.controller.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        [CONTENT_TYPE.JSON]: {
+          schema: getModelSchemaRef(RefreshTokenRequest, {
+            partial: true,
+          }),
+        },
+      },
+    })
+    req: RefreshTokenRequest,
+  ): Promise<SuccessResponse> {
+    const token = auth?.replace(/bearer /i, '');
```
Comment (samarpan-b) [inline services/authentication-service/src/modules/auth/controllers/identity-server.controller.ts:224]: Cant we reuse existing code ?

**Review by samarpan-b — APPROVED** [review]

File: packages/core/src/components/bearer-verifier/providers/services-bearer-asym-token-verifier.ts
Comment (yeshamavani) [inline packages/core/src/components/bearer-verifier/providers/services-bearer-asym-token-verifier.ts:1]: @prernagp dont we need to do similar changes to 'FacadesBearerAsymmetricTokenVerifyProvider' - packages/core/src/components/bearer-verifier/providers

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: packages/core/src/components/bearer-verifier/providers/services-bearer-asym-token-verifier.ts
Comment (prernagp) [reply packages/core/src/components/bearer-verifier/providers/services-bearer-asym-token-verifier.ts:1]: done.

File: packages/core/src/components/bearer-verifier/repositories/revoked-token.repository.ts [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
 import {DefaultKeyValueRepository, juggler} from '@loopback/repository';
 
+import {AuthDbSourceName} from '../../../types';
 import {RevokedToken} from '../models';
-import {AuthCacheSourceName} from '../types';
 
 export class RevokedTokenRepository extends DefaultKeyValueRepository<RevokedToken> {
   constructor(
-    @inject(`datasources.${AuthCacheSourceName}`)
+    @inject(`datasources.${AuthDbSourceName}`)
```
Comment (yeshamavani) [inline packages/core/src/components/bearer-verifier/repositories/revoked-token.repository.ts:13]: Looks like done by mistake 
This is a key value store - redis 
should be 'AuthCacheSourceName'

File: packages/cli/README.md
Comment (yeshamavani) [inline packages/cli/README.md:1]: Please revert this file

**Pull request merged**